### PR TITLE
Updated `atsamd-hal` to 0.15.1 for pyportal

### PR DESCRIPTION
**Pull request by wembly**
# Summary
Updated `atsamd-hal` from 0.14.0 to 0.15.1
Incremented crate version to 0.10.0
Added usb feature to bsp, and example
Added display feature to bsp and example

# Checklist
  - [x] `CHANGELOG.md` for the BSP or HAL updated
  - [x] All new or modified code is well documented, especially public items
  - [x] No new warnings or clippy suggestions have been introduced (see CI or check locally)

### CODE REVIEW

File: boards/pyportal/Cargo.toml [hunk 1]
```diff
@@ -15,26 +15,43 @@ readme = "README.md"
...
+optional=true
+
+[dependencies.usb-device]
+version = "0.2.8"
+optional = true
+
 [dev-dependencies]
 panic-halt = "0.2"
 panic-semihosting = "0.5"
 smart-leds = "~0.3"
```
Comment (sajattack) [inline boards/pyportal/Cargo.toml:40]: We might want to move this out of dev-dependencies and add some support functions to the bsp. Up to you though.

**Comment by sajattack** [comment]
Looks pretty good to me!

File: boards/pyportal/Cargo.toml (hunk 1 above)
Comment (sajattack) [reply boards/pyportal/Cargo.toml:40]: Ugh, github kinda broke my comment, I was referring to `smart-leds`

Comment (wembly) [reply boards/pyportal/Cargo.toml:40]: following up on this: we discussed this online and decided not to add this dependency at this point in time

File: boards/pyportal/src/display.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+use super::pins::*;
+use atsamd_hal::ehal::blocking::delay::DelayMs;
+use atsamd_hal::prelude::*;
+use display_interface_parallel_gpio::{Generic8BitBus, PGPIO8BitInterface};
```
Comment (sajattack) [inline boards/pyportal/src/display.rs:4]: surprised if the ili9341 crate doesn't re-export this

Comment (sajattack) [inline boards/pyportal/src/display.rs:4]: surprised if the ili9341 crate doesn't re-export this

Comment (wembly) [reply boards/pyportal/src/display.rs:4]: unfortunately it is not re-exported from what i can tell

File: boards/pyportal/src/pins.rs [hunk 3]
```diff
@@ -1,116 +1,350 @@
...
+        name: neopixel,
+        aliases: {
+            Reset: NeopixelReset,
+            PushPullOutput: Neopixel,
+        }
+    }
+
+    PA00 {
+        // TFT(Thin-film-transistor liquid-crystal display) control ports
+        /// TFT Reset
```
Comment (bradleyharden) [inline boards/pyportal/src/pins.rs:64]: A few of these have part of the documentation as a comment and part as actual docs. Is there a reason for that?

Comment (wembly) [reply boards/pyportal/src/pins.rs:64]: i carried that over from the previous version, from what i've been able to discern the comments are to indicate logical groupings of pins in the code, while the documentation is for to document the pin itself

**Comment by wembly** [comment]
> This looks good to me. Have all the examples been tested?

I've tested them all on the physical pyportal hardware i have by using `cargo run --example <example name> --release` if that's what your asking

**Comment by bradleyharden** [comment]
Yeah, that's what I was asking. I think this is good to merge then. @sajattack, are you good?

**Pull request merged**